gateway/webhook/webhookTransformer.go

```diff
@@ -10,11 +10,75 @@ import (
 	"strings"
 	"time"
 
+	"github.com/rudderlabs/rudder-go-kit/config"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	gwtypes "github.com/rudderlabs/rudder-server/gateway/internal/types"
 	"github.com/rudderlabs/rudder-server/gateway/response"
+	"github.com/rudderlabs/rudder-server/services/transformer"
 	"github.com/rudderlabs/rudder-server/utils/httputil"
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
+type sourceTransformAdapter interface {
+	getTransformerEvent(authCtx *gwtypes.AuthRequestContext, body []byte) ([]byte, error)
+	getTransformerURL(sourceType string) string
+}
+
+type v0Adapter struct{}
+
+type v1Adapter struct{}
+
+type V1TransformerEvent struct {
+	Event  json.RawMessage       `json:"event"`
+	Source backendconfig.SourceT `json:"source"`
+}
+
+func (v0 *v0Adapter) getTransformerEvent(authCtx *gwtypes.AuthRequestContext, body []byte) ([]byte, error) {
+	return body, nil
+}
+
+func (v0 *v0Adapter) getTransformerURL(sourceType string) string {
+	return getTransformerURL(transformer.V0, sourceType)
+}
+
+func (v1 *v1Adapter) getTransformerEvent(authCtx *gwtypes.AuthRequestContext, body []byte) ([]byte, error) {
+	source := authCtx.Source
+
+	v1TransformerEvent := V1TransformerEvent{
+		Event: body,
+		Source: backendconfig.SourceT{
+			ID:               source.ID,
+			OriginalID:       source.OriginalID,
+			Name:             source.Name,
+			SourceDefinition: source.SourceDefinition,
+			Config:           source.Config,
+			Enabled:          source.Enabled,
+			WorkspaceID:      source.WorkspaceID,
+			WriteKey:         source.WriteKey,
+			Transient:        source.Transient,
+		},
+	}
+
+	return json.Marshal(v1TransformerEvent)
+}
+
+func (v1 *v1Adapter) getTransformerURL(sourceType string) string {
+	return getTransformerURL(transformer.V1, sourceType)
+}
+
+func newSourceTransformAdapter(version string) sourceTransformAdapter {
+	switch version {
+	case "v1":
+		return &v1Adapter{}
+	}
+
+	return &v0Adapter{}
+}
+
+func getTransformerURL(version, sourceType string) string {
+	return strings.TrimSuffix(config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090"), "/") + "/" + version + "/sources/" + sourceType
```

Comment:
Prefer [url.JoinPath](https://pkg.go.dev/net/url@go1.19beta1#JoinPath), for readability and safety.
```suggestion
        bashURL := config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")
	return url.JoinPath(bashURL, version, "sources", sourceType)
```

You might need to use [PathEscape](https://cs.opensource.google/go/go/+/go1.19beta1:src/net/url/url.go;l=281) if your `sourceType` is not safe to be encoded in the url